src/main/java/org/plp/isa/AsmFile.java

```diff
@@ -28,25 +28,24 @@
     void setFilePath(String filePath);
 
     /**
-     *  Given a list of instructions of the program, this is add it to its inmemory representation of that program
+     *  Given a list of instructions of the program, this will add it to its inmemory representation of that program
      * @param instructions List of instructions to be added to its in-memory representation
-     * @return list of {@link AsmInstruction} which is equivalent of the string representation
      */
-    List<AsmInstruction> addInstructionsToFile(List<String> instructions);
+    void addInstructionsToFile(List<String> instructions);
```

Comment:
int addInstructionsToFile(List<String> instructions)

return the number of instructions has been successfully added to the inmemory representation.

Or return a boolean? I think the caller needs to know the calling is successful or not.

Reply:
I think boolean is a good fit for return value as just returning number of instructions successfully added needs to be again checked on the caller side to verify everything added or not. I will make return type as boolean.

Reply:
In this case, if we are using `boolean` I would suggest 

```
boolean addInstructionToFile(String instruction);
```

And the caller needs to iterate through the List<> and handle the return value.

Reply:
Yes we can do that way.